ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/CIOEngine.kt

```diff
@@ -59,10 +79,7 @@ internal class CIOEngine(override val config: CIOEngineConfig) : HttpClientEngin
     override fun close() {
         super.close()
         endpoints.forEach { (_, endpoint) -> endpoint.close() }
-
-        coroutineContext[Job]!!.invokeOnCompletion {
-            selectorManager.close()
-        }
+        (requestsJob[Job] as CompletableJob).complete()
```

Comment:
❓ Shouldn't we do it in a finally block to ensure completion? Do we know that close will never crash?

Reply:
I'll add a test with manual cancellation